instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -135,10 +135,14 @@ func (h *clientHandler) HandleConn(context.Context, stats.ConnStats) {
 
 func (c *config) handleRPC(ctx context.Context, rs stats.RPCStats, isServer bool) { // nolint: revive  // isServer is not a control flag.
 	span := trace.SpanFromContext(ctx)
-	gctx, _ := ctx.Value(gRPCContextKey{}).(*gRPCContext)
+	var metricAttrs []attribute.KeyValue
 	var messageId int64
-	metricAttrs := make([]attribute.KeyValue, 0, len(gctx.metricAttrs)+1)
-	metricAttrs = append(metricAttrs, gctx.metricAttrs...)
+
+	gctx, ok := ctx.Value(gRPCContextKey{}).(*gRPCContext)
+	if ok && gctx != nil {
+		metricAttrs = make([]attribute.KeyValue, 0, len(gctx.metricAttrs)+1)
+		metricAttrs = append(metricAttrs, gctx.metricAttrs...)
```

Comment:
You can use `copy` instead of you replace the zero on the preceding line with the length of the source slice. Alternately, consider using [the `slices.Clone` function](https://pkg.go.dev/golang.org/x/exp/slices#Clone) if you don't really need that extra slot in the backing array. It's implemented by `append` today, without the preceding call to `make`.